universal-application-tool-0.0.1/app/forms/BlockVisibilityPredicateForm.java

```diff
@@ -0,0 +1,99 @@
+package forms;
+
+import play.data.validation.Constraints;
+import play.data.validation.Constraints.Validatable;
+import services.applicant.question.Scalar;
+import services.program.predicate.Operator;
+
+public class BlockVisibilityPredicateForm implements Validatable<String> {
+  @Constraints.Required(message = "Must select 'hidden if' or 'shown if'.")
```

Comment:
whoa! I've never seen this validation pattern, it's really cool. We should probably add this in our other forms, too

Reply:
I know... I found this a few weeks ago. Was excited to finally use it 😅  Another cool thing is that you can put something like `message = "error.operatorRequired" that references `conf/messages`! However, I wasn't sure how to parameterize it / how to pass in the args for a parameterized message here, so I just did it like this for now.